src/lib/chart/dendrogram.typ

```diff
@@ -99,33 +90,53 @@
 
   assert(mode in dendrogram-modes,
     message: "Invalid dendrogram mode. Use: " + repr(dendrogram-modes))
-  assert(type(x1-key) in (int, str))
-  assert(type(x2-key) in (int, str))
-  assert(type(height-key) in (int, str))
-
-  if size.at(0) == auto {
-    size.at(0) = (data.len() + 2)
+  assert(type(x1-key) in (int, str),
+    message: "Invalid x1-key type. Must be a integer or a string")
+  assert(type(x2-key) in (int, str),
+    message: "Invalid x2-key type. Must be a integer or a string")
+  assert(type(height-key) in (int, str),
+    message: "Invalid height-key type. Must be a integer or a string")
+
+  if ( size == auto ){
+    if mode == "vertical" {size = (auto, 1)} 
+    if mode == "horizontal" {size = (1, auto)}
   }
 
-  let max-value = (dendrogram-max-value-fn.at(mode))(data, height-key)
-  if y-max != auto {
-    max-value = y-max
-  }
-  let min-value = (dendrogram-min-value-fn.at(mode))(data, height-key)
-  if y-min != auto {
-    min-value = y-min
+  let size-node-axis-index = 0
+  if ( mode == "horizontal" ){size-node-axis-index = 1}
+
+  if size.at(size-node-axis-index) == auto {
+    size.at(size-node-axis-index) = (data.len() + 2)
   }
 
-  let x-ticks = x-ticks
+  let max-value = dendrogram-max-value-fn(data, height-key)
+  if y-max != auto { max-value = y-max }
+
+  let min-value = dendrogram-min-value-fn(data, height-key)
+  if y-min != auto { min-value = y-min }
```

Comment:
These could be `if...else` statements. I think calculating the max/min in discrete functions doesn't help code readability as they aren't used anywhere else and are only one line.

Reply:
done